olga/analytics/views.py

```diff
@@ -2,38 +2,111 @@
 Views for the analytics application.
 """
 
+import httplib
 import json
+import logging
 import uuid
 
-import requests
-
-from django.conf import settings
 from django.http import HttpResponse
+from django.http import JsonResponse
 from django.views.generic import View
 from django.views.decorators.csrf import csrf_exempt
 from django.utils.decorators import method_decorator
 
+from .forms import AccessTokenForm
 from .models import InstallationStatistics, EdxInstallation
+from .utils import installation_statistics_forms_checker
+
+logging.basicConfig()
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
 
-HTTP_201_CREATED = 201
+stream_handler = logging.StreamHandler()
+stream_handler.setLevel(logging.INFO)
 
 
 @method_decorator(csrf_exempt, name='dispatch')
-class ReceiveData(View):
+class AccessTokenRegistration(View):
     """
-    Receives and processes data from the remote edx-platform.
+    Provide access token registration functionality.
+    """
+
+    @staticmethod
+    def registry_a_token_and_return_it():
+        """
+        Create UUID based access token ans store it in database.
+
+        Return same access token.
+        """
+        access_token = uuid.uuid4().hex
+        EdxInstallation.objects.create(access_token=access_token)
 
-    If the platform has already registered, a normal data exchange happens.
-    Otherwise generates a secret token, (registers)saves it to DB with the edx-platform's incoming URL
-    and sends newly generated token to the edx-platform for further
-    data interchange abilities with the server.
+        return access_token
+
+    def post(self, request):  # pylint: disable=unused-argument
+        """
+        Receive primary edX installation request for getting access for dispatch statistics via token.
+
+        Returns HTTP-response with status 201, that means object (installation token) was successfully created.
+        """
+        access_token = self.registry_a_token_and_return_it()
+
+        logger.info('OLGA acceptor registered edX installation with token %s', access_token)
+        return JsonResponse({'access_token': access_token}, status=httplib.CREATED)
+
+
+@method_decorator(csrf_exempt, name='dispatch')
+class AccessTokenAuthorization(View):
+    """
+    Provide access token authorization functionality.
+    """
+
+    @staticmethod
+    def post(request):
```

Comment:
Split this view into two views.

1. GET - `AuthorizeAccessToken` - To check if API consumer has access to the system.
2. POST - `RegisterEdxInstance` - To register new edX instance.